platform/security/handler/security-handler-oauth/src/main/java/org/codice/ddf/security/handler/oauth/CustomOAuthCredentialsExtractor.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.handler.oauth;
+
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.utils.OAuthEncoder;
+import org.pac4j.core.client.IndirectClient;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.oauth.config.OAuth20Configuration;
+import org.pac4j.oauth.credentials.OAuth20Credentials;
+import org.pac4j.oauth.credentials.extractor.OAuth20CredentialsExtractor;
+
+public class CustomOAuthCredentialsExtractor extends OAuth20CredentialsExtractor {
```

Comment:
📘 This class's job is to extract an OAuth2 `AccessToken` from either the query parameters or the https headers.